Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -3012,150 +3012,160 @@ RefPtr<StyleValue> Parser::parse_identifier_value(ComponentValue const& componen
-                    return Color(r, g, b, a);
-
-            } else if (r_val.is(Token::Type::Percentage)
-                && g_val.is(Token::Type::Percentage)
-                && b_val.is(Token::Type::Percentage)
-                && a_val.is(Token::Type::Number)) {
-
-                auto r = r_val.percentage();
-                auto g = g_val.percentage();
-                auto b = b_val.percentage();
-                auto a = a_val.number_value();
-
-                u8 r_255 = clamp(lroundf(r * 2.55f), 0, 255);
-                u8 g_255 = clamp(lroundf(g * 2.55f), 0, 255);
-                u8 b_255 = clamp(lroundf(b * 2.55f), 0, 255);
-                u8 a_255 = clamp(lroundf(a * 255.0f), 0, 255);
-                return Color(r_255, g_255, b_255, a_255);
-            }
-        } else if (function.name().equals_ignoring_case("hsl")) {
-            if (params.size() != 3)
+            float numeric_value = params[0].dimension_value();
+            auto unit_string = params[0].dimension_unit();
+            auto angle_type = Angle::unit_from_name(unit_string);
+
+            if (!angle_type.has_value())
                 return {};
 
-            auto h_val = params[0];
-            auto s_val = params[1];
-            auto l_val = params[2];
+            auto angle = Angle { numeric_value, angle_type.release_value() };
 
-            if (h_val.is(Token::Type::Number)
-                && s_val.is(Token::Type::Percentage)
-                && l_val.is(Token::Type::Percentage)) {
+            float h_val = fmodf(angle.to_degrees(), 360.0f);
+            float s_val = params[1].percentage() / 100.0f;
+            float l_val = params[2].percentage() / 100.0f;
 
-                auto h = h_val.number_value();
-                auto s = s_val.percentage() / 100.0f;
-                auto l = l_val.percentage() / 100.0f;
-                return Color::from_hsl(h, s, l);
-            }
-        } else if (function.name().equals_ignoring_case("hsla")) {
-            if (params.size() != 4)
-                return {};
+            return Color::from_hsla(h_val, s_val, l_val, a_val);
+        }
 
-            auto h_val = params[0];
-            auto s_val = params[1];
-            auto l_val = params[2];
-            auto a_val = params[3];
-
-            if (h_val.is(Token::Type::Number)
-                && s_val.is(Token::Type::Percentage)
-                && l_val.is(Token::Type::Percentage)
-                && a_val.is(Token::Type::Number)) {
-
-                auto h = h_val.number_value();
-                auto s = s_val.percentage() / 100.0f;
-                auto l = l_val.percentage() / 100.0f;
-                auto a = a_val.number_value();
-                return Color::from_hsla(h, s, l, a);
-            }
+        if (params[0].is(Token::Type::Number)
+            && params[1].is(Token::Type::Percentage)
+            && params[2].is(Token::Type::Percentage)) {
+
+            float h_val = fmodf(params[0].number_value(), 360.0f);
```

Comment:
Same as above. Also, the two branches could be combined since only the code for getting the hue is different, but keeping them separate is fine too. :^)

Reply:
I tried to match the existing code layout which was quite readable. :^)